### DISCUSSION TITLE
Unable to run DIP.ipynb

### DISCUSSION THREAD

**Discussion by ParamB11** [do-mpc/do-mpc#255] [Q&A]
I have downloaded the notebook from the do-mpc website and am trying to run the DIP.ipynb but I am getting the following error. 

```
RuntimeError                              Traceback (most recent call last)
<ipython-input-27-819c23d25edf> in <module>
      5 for k in range(n_steps):
      6     u0 = mpc.make_step(x0)
----> 7     y_next = simulator.make_step(u0)
      8     x0 = estimator.make_step(y_next)

G:\XYZ\Software\Anaconda\lib\site-packages\do_mpc\simulator.py in make_step(self, u0, v0, w0)
    498         self.sim_p_num['_w'] = w0
    499 
--> 500         x_next = self.simulate()
    501 
    502         z0 = self.sim_z_num['_z']

G:\XYZ\Software\Anaconda\lib\site-packages\do_mpc\simulator.py in simulate(self)
    429             x_new = self.simulator(sim_x_num, sim_z_num, sim_p_num)
    430         elif self.model.model_type == 'continuous':
--> 431             r = self.simulator(x0 = sim_x_num, z0 = sim_z_num, p = sim_p_num)
    432             x_new = r['xf']
    433             z_now = r['zf']

G:\XYZ\Software\Anaconda\lib\site-packages\casadi\casadi.py in __call__(self, *args, **kwargs)
  13451       else:
  13452     # Named inputs -> return dictionary
> 13453         return self.call(kwargs)
  13454 
  13455     def buffer(self):

G:\XYZ\Software\Anaconda\lib\site-packages\casadi\casadi.py in call(self, *args)
  12322 
  12323         """
> 12324         return _casadi.Function_call(self, *args)
  12325 
  12326 

RuntimeError: .../casadi/interfaces/sundials/idas_interface.cpp:591: IDACalcIC returned "IDA_LINESEARCH_FAIL". Consult IDAS documentation.
```

Can you please tell how to solve the error. I haven't changed anything in the downloaded notebook.

**Comment by 4flixt** [comment]
Hi, @ParamB11 ,

I have just opened the Notebook through Binder (directly from the website) and the code ran fine for me. It might be a problem with CasADi and your python installation. CasADi does have problems with Python 3.10 for example, see [here](https://github.com/casadi/casadi/issues/2877). 

Maybe try creating a new (anaconda) environment with Python 3.6. Sometimes it also helps to install CasADi from pip instead through conda. 

Best,
4flixt

**Reply by ParamB11** [reply]
I am currently using python version 3.8.5 .
`Python version in use:  3.8.5 (default, Sep  3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)]`

I installed do-mpc using pip and during that process CasADi was installed probably using pip.
So, do I need to install CasADi once again using pip?

**Comment by drewhamiltonasdf** [comment]
I had the same issue. I changed the time-step from 0.04 to 0.038 with time horizon of 110 steps and it solved the closed-loop portion. Presumably issues with IPOPT between versions not finding the same solution?

However, the DIP unfortunately does not stabilize to the upright position.

I had to increase the max force to [-5, 5] and bump n_steps up to 160.

**Comment by 4flixt** [comment]
I stumbled upon this issue again and found that IDAS is very sensitivity to the parameters ``rtol`` and ``abstol`` for this particular DAE system.

For me this worked:
```
    params_simulator = {
        # Note: cvode doesn't support DAE systems.
        'integration_tool': 'idas',
        'abstol': 1e-8,
        'reltol': 1e-8,
        't_step': 0.04
    }
```
